### PR TITLE
Lucene 8.11.1 re-introduced a lot of checksum issues #1567

### PR DESCRIPTION
**Pull request by jleach4**
8.11.1 re-introduced checksums that we had attempted to remove in our prior version.

### CODE REVIEW

File: fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java [hunk 1]
```diff
@@ -0,0 +1,2006 @@
...
+    private static final long BASE_RAM_BYTES_USED = RamUsageEstimator.shallowSizeOfInstance(Lucene84PostingsReader.class);
+
+    private final IndexInput docIn;
+    private final IndexInput posIn;
+    private final IndexInput payIn;
+
+    private final int version;
+
+    /** Sole constructor. */
+    public LuceneOptimizedPostingsReader(SegmentReadState state) throws IOException {
```
Comment (tian-yizuo) [inline fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java:76]: Could you add comments to indicate that the difference between this optimized one and the ```Lucene84PostingsReader``` is the removed ```CodecUtil.retrieveChecksum```

**Review by tian-yizuo — COMMENTED** [review]
LGTM basically, just left one minor comment.

File: fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java [hunk 2]
```diff
@@ -0,0 +1,2006 @@
+/*
+ * LuceneOptimizedPostingsReader.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
```
Comment (alecgrieser) [inline fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java:6]: If this is taken from code that was originally in Lucene and then modified, I think we need a slightly different attribution header here. I thought there was an example somewhere of a place we've done this, but I'm not actually sure that there is

File: fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/package-info.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Support for {@code LUCENE} indexes and queries.
```
Comment (alecgrieser) [inline fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/package-info.java:22]: This looks like it's been taken from the `com.apple.foundationdb.record.lucene` package-info. This should probably describe the code in this subpackage specifically, including noting that the codec is adapted from the 8.11 optimized codec

**Review by alecgrieser — CHANGES_REQUESTED** [review]
In addition to the comments left in-line, it might also be helpful to update the release notes. I don't know if the Lucene work has been using them, but I (personally, at least) find those are pretty useful when upgrading the Record Layer dependency so that it's clearer what the differences between the versions are

File: fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java (hunk 2 above)
Comment (jleach4) [reply fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java:6]: Could not find it..

**Review by alecgrieser — DISMISSED** [review]

**Comment by alecgrieser** [comment]
It looks like the build failed because checkstyle noticed that the new file has a non-standard header:

```
[ant:checkstyle] [ERROR] /codebuild/output/src566016129/src/github.com/FoundationDB/fdb-record-layer/fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java:3: Line does not match expected header line of ' *'. [Header]
```

But that's what we want in this case because we want the Lucene header, not our header.

I think we can suppress this check for this file by adding the following to `gradle/codequality/suppressions.xml`:

```xml
    <suppress checks="Header"
        files=".*[\\/\]org[\\/]apache[\\/]lucene[\\/]codecs[\\/]lucene84[\\/]LuceneOptimizedPostingsReader\.java"/>
```

In theory, we could suppress all checks (that is `checks=".*"`) if we don't want to run our style checks on this code.

**Review by tian-yizuo — APPROVED** [review]

**Review by tian-yizuo — DISMISSED** [review]

File: gradle/codequality/suppressions.xml [hunk 4]
```diff
@@ -8,4 +8,6 @@
         files=".*[\\/]protogen[\\/].*"/>
     <suppress checks="JavadocPackage"
         files=".*[\\/]test[\\/].*|.*[\\/]java-proto[23][\\/].*[\\/]metadata[\\/].*"/>
+    <suppress checks="Header"
+              files=".*[\\/\]org[\\/]apache[\\/]lucene[\\/]codecs[\\/]lucene84[\\/]LuceneOptimizedPostingsReader\.java"/>
```
Comment (alecgrieser) [inline gradle/codequality/suppressions.xml:12]:
```suggestion
        files=".*[\\/]org[\\/]apache[\\/]lucene[\\/]codecs[\\/]lucene84[\\/]LuceneOptimizedPostingsReader\.java"/>
```

It looks like there's an extra `/` before `org`, which is what caused the gradle check style build to fail

**Review by alecgrieser — CHANGES_REQUESTED** [review]

**Review by alecgrieser — APPROVED** [review]

**Review by alecgrieser — APPROVED** [review]

**Review by MMcM — APPROVED** [review]

**Pull request merged**